### PR TITLE
Feature/input param file

### PR DESCRIPTION
**Pull request by laetitia-m**
Allow to have the parameter file as argument using the option `-f filename.mmg<X>` where `<X>` is `2d`, `3d` or `s`.

### CODE REVIEW

File: src/common/API_functions.c [hunk 1]
```diff
@@ -222,6 +222,34 @@ int MMG5_Set_inputSolName(MMG5_pMesh mesh,MMG5_pSol sol, const char* solin) {
   return 1;
 }
 
+/**
+ * \param mesh pointer toward the mesh structure.
+ * \param fparamin name of the input solution file.
+ * \return 1 if success, 0 if fail
+ *
+ * Set the name of input solution file.
```
Comment (Algiane) [inline src/common/API_functions.c:230]: Update Doxygen doc (wrong doc copy-paste ;-)). `solution` has to be replaced by `parameter`.

File: src/common/libmmgcommon_private.h [hunk 2]
```diff
@@ -130,6 +130,23 @@ LIBMMG_CORE_EXPORT int  MMG5_Set_outputMeshName(MMG5_pMesh mesh, const char* mes
  *
  */
 LIBMMG_CORE_EXPORT int  MMG5_Set_inputSolName(MMG5_pMesh mesh,MMG5_pSol sol, const char* solin);
+/**
+ * \param mesh pointer toward the mesh structure.
+ * \param fparamin name of the input parameter file.
+ * \return 0 if failed, 1 otherwise.
+ *
+ *  Set the name of output solution file.
+ *
```
Comment (Algiane) [inline src/common/libmmgcommon_private.h:139]: Update Doxygen doc (wrong doc copy-paste ;-)). `solution` has to be replaced by `parameter`.

File: src/common/libmmgcommon_private.h [hunk 3]
```diff
@@ -130,6 +130,23 @@ LIBMMG_CORE_EXPORT int  MMG5_Set_outputMeshName(MMG5_pMesh mesh, const char* mes
...
+ * \remark Fortran interface:
+ * >   SUBROUTINE MMG5_SET_OUTPUTSOLNAME(mesh,fparamin,strlen,retval)\n
```
Comment (Algiane) [inline src/common/libmmgcommon_private.h:141]: Wrong doc copy-paste ;-)). `OUTPUTSOL` has to be replaced by `INPUTPARAM`.

Updating this specific part of doc really matters: it is used to generated automatically the Fortran interface.

File: src/mmg2d/libmmg2d.h [hunk 4]
```diff
@@ -219,6 +219,23 @@ extern "C" {
  *
  */
   LIBMMG2D_EXPORT int  MMG2D_Set_outputSolName(MMG5_pMesh mesh,MMG5_pSol sol, const char* solout);
+/**
+ * \param mesh pointer toward the mesh structure.
+ * \param fparamin name of the input parameter file.
+ * \return 1.
+ *
+ * Set the name of input solution file.
+ *
```
Comment (Algiane) [inline src/mmg2d/libmmg2d.h:228]: Solution -> parameter

File: src/mmg3d/API_functionsf_3d.c [hunk 5]
```diff
@@ -137,6 +137,24 @@ FORTRAN_NAME(MMG3D_SET_OUTPUTSOLNAME,mmg3d_set_outputsolname,
...
+FORTRAN_NAME(MMG3D_SET_INPUTPARAMNAME, mmg3d_set_inputparamname,
+             (MMG5_pMesh *mesh,char* fparamin, int* strlen0, int* retval),
+             (mesh,fparamin,strlen0,retval)) {
+
+  char *tmp = NULL;
+
+  MMG5_SAFE_MALLOC(tmp,*strlen0+1,char,return);
+  strncpy(tmp,fparamin,*strlen0);
+  tmp[*strlen0] = '\0';
+  *retval = MMG3D_Set_inputParamName(*mesh,tmp);
```
Comment (Algiane) [inline src/mmg3d/API_functionsf_3d.c:152]: No need to change this but MMG5_Set_inputParaName can be called directly (it has the same result but if someone searches what is doing this functions, it allows to access the code source faster (you have one less redirection).
If the function was called a huge amout of time (not the case here), it is also possible that adding one useless function call will slightly slow down the code.

File: src/mmg3d/API_functionsf_3d.c [hunk 6]
```diff
@@ -137,6 +137,24 @@ FORTRAN_NAME(MMG3D_SET_OUTPUTSOLNAME,mmg3d_set_outputsolname,
   return;
 }
 
+/**
+ * See \ref MMG3D_Set_inputParamName function in \ref common/libmmgcommon_private.h file.
+ */
```
Comment (Algiane) [inline src/mmg3d/API_functionsf_3d.c:142]: I guess that the documentation issue comes from an error for another function but `MMG3D_Set_inputParamName` is declared in `mmg3d/libmmg3d.h` file.

**Comment by Algiane** [comment]
Thanks for this PR!

File: src/mmg3d/API_functionsf_3d.c (hunk 5 above)
Comment (laetitia-m) [reply src/mmg3d/API_functionsf_3d.c:152]: Noted for future PR. Thanks.

**Comment by Algiane** [comment]
Thanks again, it is merged!

**Pull request merged**